worker/uniter/runner/runner.go

```diff
@@ -312,30 +312,19 @@ func (runner *runner) updateActionResults(results *utilexec.ExecResponse) error
 }
 
 // RunAction exists to satisfy the Runner interface.
-func (runner *runner) RunAction(actionName string) error {
+func (runner *runner) RunAction(actionName string) (HookHandlerType, error) {
 	if _, err := runner.context.ActionData(); err != nil {
-		return errors.Trace(err)
+		return InvalidHookHandler, errors.Trace(err)
 	}
 	if actionName == actions.JujuRunActionName {
-		return runner.runJujuRunAction()
+		return InvalidHookHandler, runner.runJujuRunAction()
```

Comment:
Just looking at this, I know it's merged but here could we return a different value for juju run action. Possibly the handler could be called ExecHandler.